### ISSUE TITLE
Allow to configure TLS properties

### ISSUE THREAD

**Issue by LazyAspect** [AdguardTeam/AdGuardHome#1207]
Hello,

I would like to be able to control the cipher types for **DoT**. For example, I would like to use only these cipher types: **ECDH+AESGCM:EECDH+CHACHA20**.

It would be great to have an appropriate flag entry field for this or something like this.

Thank you.

**Comment by ameshkov** [comment]
I reckon you'd like to configure the minimum TLS version as well?

**Comment by LazyAspect** [comment]
@ameshkov, I think using the default maximum TLS version is sufficient. But control of cipther types will not interfere.

**Comment by ameshkov** [comment]
Well, cipher suites and TLS versions are connected. `ECDH+AESGCM:EECDH+CHACHA20` simply aren't available if you disable TLS 1.2 and older versions.

**Comment by LazyAspect** [comment]
@ameshkov, yeah, I know. It may make sense to use predefined combinations of TLS versions and cipher types.

**Comment by ammnt** [comment]
@ameshkov, I have an alternative solution in the context of this task. In order not to make this difficult, I suggest to completely disable TLS 1.0 and TLS 1.1 (or even later...) support for encrypted protocols and upstream servers, or make it optional via yaml. Then we can be sure that vulnerable versions of protocols are not used. What do you think?🤔

**Comment by ameshkov** [comment]
Hmm, it is possible that we have already done this, could you please check it?

**Comment by ammnt** [comment]
> Hmm, it is possible that we have already done this, could you please check it?

Yeah, no support of TLS 1.0 and 1.1 for encrypted protocols. But I don't know how to test this for upstream-servers🙄

**Comment by ameshkov** [comment]
I checked the source code and we do set min version to TLS 1.2
![image](https://user-images.githubusercontent.com/5947035/85338392-e1b05c80-b4ea-11ea-92ba-579e451cf6c9.png)

**Comment by ammnt** [comment]
@ameshkov, awesome!😙

**Comment by jyxjjj** [comment]
> I checked the source code and we do set min version to TLS 1.2


I think it is good to be a configuration.
I need TLS1.3 Only. Sometimes, maybe ESNI only.

**Comment by sofvaskina** [comment]
@jyxjjj Hello! Please create a new feature request.

**Comment by sofvaskina** [comment]
I am closing this request as resolved.
